### PR TITLE
create the add event in admin panel relates #5

### LINKED ISSUE #5: Add event
## Acceptance criteria : 
- [ ] add event with the required details(title, description, time, etc)
- [ ] choose to post to the Facebook

### PR DESCRIPTION
**Pull request by amusameh**
Fixes
fixes #5

Related
related to #5 and #7 

Summary
Create the add event page for the admin panel, and insert into the database.
Upload images using the express-fileupload module.
Add course

### CODE REVIEW

**Comment by mohanagy** [comment]
fix conflict

File: public/css/admin.css [hunk 1]
```diff
@@ -328,3 +390,4 @@ background-color:#fff;
   border-radius: 25px;
   margin: 20px auto;
 }
+>>>>>>> bdcdd313d033ac5ecf59da2f044365ac9d56ff48
```
Comment (mohanagy) [inline public/css/admin.css]: WTF

File: public/css/admin.css [hunk 2]
```diff
@@ -264,6 +324,8 @@ background-color:#F4F4F4;
 .tableElements:nth-of-type(even) {
 background-color:#fff;
 }
+<<<<<<< HEAD
```
Comment (mohanagy) [inline public/css/admin.css]: WTF

File: src/controllers/index.js [hunk 3]
```diff
@@ -20,5 +21,7 @@ router
   .get('/admin/events', adminEvents)
   .get('/admin/courses', adminCourses)
   .get('/courses', events)
+  .get('/admin/addEventsPage', getAdminPage)
+  .post('/admin/addevent', postEvent)
```
Comment (mohanagy) [inline src/controllers/index.js]: we need validation

File: public/css/admin.css (hunk 1 above)
Comment (amusameh) [reply public/css/admin.css]: conflicts hell

File: src/app.js [hunk 4]
```diff
@@ -3,6 +3,7 @@ import bodyParser from 'body-parser';
 import exhbs from 'express-handlebars';
 import path from 'path';
 import favicon from 'serve-favicon';
+import fileUpload from 'express-fileupload';
 import controllers from './controllers';
 import helpers from './views/helpers/index';
```
[the remark this answers is not in the document]
Comment (amusameh) [reply src/app.js]: edited

File: src/controllers/adminAddCourse.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+};
+
+export const postCourse = (req, res, next) => {
+  const {
+    courseTitle, courseDiscription, courseFees,
+  } = req.body;
+  const { courseImage } = req.files;
+  if (courseTitle && courseDiscription && courseFees) {
+    uploadPhoto(courseImage, next, (photo) => {
+      const imgLink = photo.url;
```
[the remark this answers is not in the document]
Comment (amusameh) [reply src/controllers/adminAddCourse.js]: Edited
What if i want to rename the variable, should it be like this??
const{ url: imgLink } = photo

File: src/controllers/adminAddEvent.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  });
+};
+
+export const postEvent = (req, res, next) => {
+  const {
+    eventTitle, eventDate, startTime, endTime, eventPlace, eventDiscription,
+  } = req.body;
+  const { eventImage } = req.files;
+
+  if (eventTitle && eventDate && startTime && endTime && eventPlace
```
[the remark this answers is not in the document]
Comment (amusameh) [reply src/controllers/adminAddEvent.js]: I did use it as a middleware then applied it to the routes. same as for the login 

What is the right way to use it?

File: src/controllers/validate_course_event.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
+import Joi from 'joi';
+
+const schema = Joi.object().keys({
```
[the remark this answers is not in the document]
Comment (amusameh) [reply src/controllers/validate_course_event.js]: Yes, it works for me.

what if we want to use joi with params and body at the same time?
I don't understand, Explain more, please

**Pull request merged**